### PR TITLE
search functionality added

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/views/products/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,19 @@
+
 <%= @current_user  %>
 <h4>This is the product page</h4>
+<h5> Search for Products Below: </h5>
+<%= form_tag(products_path, :method => "get", id: "search-form") do %>
+<%= text_field_tag :search, params[:search], placeholder: "Search Products" %>
+<%= submit_tag "Search" %>
+<% end %>
```
Comment (radavis) [inline app/views/products/index.html.erb:9]: This could be formatted better.

File: app/views/products/index.html.erb [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+
+<% if @products.present? %>
+<% else %>
```
Comment (radavis) [inline app/views/products/index.html.erb]: You are expressing in two lines what could easily be expressed in one line:

```
if @products.empty?
```

or

```
if !@products.present?
```

or 

```
unless @products.present?
```

File: app/views/products/index.html.erb [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+  <p>There are no posts containing the term(s) <%= params[:search] %>.</p>
+<% end %>
+
+  Product Index:
+  <br>
```
Comment (radavis) [inline app/views/products/index.html.erb]: You probably want to wrap this in a header tag instead of using a `<br />`

**Comment by radavis** [comment]
No tests?

I need to see a unit test for the search feature, at a minimum.
